From 4697a8954ba03c165fbc2e06015a9ae87328a508 Mon Sep 17 00:00:00 2001
From: Alice Ryhl <aliceryhl@google.com>
Date: Tue, 4 Nov 2025 14:14:22 +0000
Subject: [PATCH 151/161] tyr: do not hardcode vm_operations_struct fields

This does not have the same fields under all configs, so do not
hard-code the list as it fails to compile if you get it wrong.

Signed-off-by: Alice Ryhl <aliceryhl@google.com>
---
 drivers/gpu/drm/tyr/mmap.rs | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/drivers/gpu/drm/tyr/mmap.rs b/drivers/gpu/drm/tyr/mmap.rs
index 254b787d51..61216b4dc7 100644
--- a/drivers/gpu/drm/tyr/mmap.rs
+++ b/drivers/gpu/drm/tyr/mmap.rs
@@ -56,22 +56,9 @@ pub(crate) fn mmap(device: &TyrDevice, _file: &crate::file::File, vma: &VmaNew)
 }
 
 static VM_OPS: bindings::vm_operations_struct = bindings::vm_operations_struct {
-    open: None,
-    close: None,
-    may_split: None,
-    mremap: None,
-    mprotect: None,
     fault: Some(vm_fault_handler),
-    huge_fault: None,
-    map_pages: None,
-    pagesize: None,
-    page_mkwrite: None,
-    pfn_mkwrite: None,
-    access: None,
-    name: None,
-    set_policy: None,
-    get_policy: None,
-    find_special_page: None,
+    // SAFETY: All zeros is valid for vm ops.
+    ..unsafe { core::mem::zeroed() }
 };
 
 unsafe extern "C" fn vm_fault_handler(vmf: *mut bindings::vm_fault) -> bindings::vm_fault_t {
-- 
2.51.0

